src/coreComponents/mesh/NodeManager.cpp

```diff
@@ -105,6 +105,44 @@ void NodeManager::buildSets( CellBlockManagerABC const & cellBlockManager,
   } );
 }
 
+void NodeManager::setIsExternal( FaceManager const & faceManager )
+{
+  // get the "isExternal" field from the faceManager...This should have been
+  // set already!
+  arrayView1d< integer const > const & isExternalFace = faceManager.isExternal();
+
+  ArrayOfArraysView< localIndex const > const & faceToNodes = faceManager.nodeList().toViewConst();
+
+  // get the "isExternal" field from for *this, and set it to zero
+  m_isExternal.zero();
+
+  // get externalSet reference to insert external faces
+  SortedArray< localIndex > & externalSet = this->externalSet();
+
+  //get normals to all faces
+  const arrayView2d< real64 const > normalOfAllFaces = faceManager.faceNormal();
+
+  // loop through all faces
+  for( localIndex kf=0; kf<faceManager.size(); ++kf )
```

Comment:
this should be a `forAll<parallelHostPolicy>`. careful not to capture member variables though.